cinema/views.py

```diff
@@ -48,11 +81,59 @@
     queryset = MovieSession.objects.all()
     serializer_class = MovieSessionSerializer
 
+    def get_queryset(self):
+        queryset = MovieSession.objects.all()
+        date = self.request.query_params.get("date")
+        movie_id = self.request.query_params.get("movie")
+
+        if date:
+            queryset = queryset.filter(show_time__date=date)
+        if movie_id:
+            queryset = queryset.filter(movie__id=int(movie_id))
+
+        if self.action == "list":
+            queryset = queryset.select_related(
+                "movie",
+                "cinema_hall"
+            ).annotate(
+                tickets_available=F(
+                    "cinema_hall__rows"
+                ) * F(
+                    "cinema_hall__seats_in_row"
+                ) - Count(
+                    "tickets"
+                )
+            )
+        return queryset.distinct()
+
     def get_serializer_class(self):
         if self.action == "list":
             return MovieSessionListSerializer
-
         if self.action == "retrieve":
             return MovieSessionDetailSerializer
-
         return MovieSessionSerializer
+
+
+class OrderViewSet(viewsets.ModelViewSet):
+    queryset = Order.objects.all()
+    serializer_class = OrderSerializer
+    pagination_class = OrderPagination
+
+    def get_queryset(self):
+        if self.request.user.is_authenticated:
+            queryset = self.queryset.filter(user=self.request.user)
+            if self.action == "list":
+                queryset = queryset.prefetch_related(
+                    "tickets__movie_session__movie"
+                )
+        else:
+            queryset = self.queryset.none()
+        return queryset
```

Comment:
The `get_queryset` method in `OrderViewSet` currently returns an empty queryset for unauthenticated users. This is correct if you want to restrict access, but ensure this behavior aligns with your application's requirements.